### ISSUE TITLE
Use -32bit packages as makedepens package

### ISSUE THREAD

**Issue by thypon** [voidlinux/void-packages#1584]
```
# Template file for 'pipelight'
pkgname=pipelight
version=0.2.8.1
revision=1
only_for_archs="i686 x86_64"
build_style=gnu-configure
configure_args=" --win32-cxx wineg++"
hostmakedepends="git wine"
makedepends="libX11-devel MesaLib-devel wine-devel"
short_desc="Windows plugins in Linux browsers"
maintainer="Andrea Brancaleoni <miwaxe@gmail.com>"
license="GPL-2"
homepage="http://pipelight.net/"

if [[ ${XBPS_TARGET} == "x86_64" ]]; then
  hostmakedepends+="-32bit"
  makedepends+="-32bit"
fi

do_fetch() {
	local url="https://bitbucket.org/mmueller2012/${pkgname}/"
	msg_normal "Fetching source from $url ...\n"
	git clone --branch v${version} ${url} ${pkgname}-${version}
}
post_install() {
  rm ${DESTDIR}/usr/share/pipelight/wine
}
```

In this scaffold i use wineg++ as a migw32 replacement. Unfortunatly wineg++ is contained in wine-32bit that can't be compiled in x86_64.

**Comment by beefcurtains** [comment]
Building gcc with x86_64-w64-mingw32 target, will try packaging it. The gcc-mingw32-w64 cross compiler would be a useful addition imho.

**Comment by thypon** [comment]
Yep, currently there is winegcc/wineg++ that's more then enough to compile this package (given we run on wine however).

**Comment by xtraeme** [comment]
https://github.com/voidlinux/void-packages/blob/master/Manual.md#32bit-packages

**Comment by thypon** [comment]
This wont solve. I need a 32bit package/library in 64bit makedepends/hostmakedepends; there is a similar problem if we want to compile wine for 64bit (with 32bit support).

**Comment by thypon** [comment]
I have the same problem with gcc-multilib:

```
$ ./xbps-src pkg gcc-multilib
=> Using `/var/packages/hostdir/binpkgs/pipelight' as local repository.
[*] Updating `http://repo.voidlinux.eu/current/x86_64-repodata' ...
x86_64-repodata: 791KB [avg rate: 112KB/s]
[*] Updating `http://repo.voidlinux.eu/current/nonfree/x86_64-repodata' ...
[*] Updating `http://repo.voidlinux.eu/current/multilib/x86_64-repodata' ...
[*] Updating `http://repo.voidlinux.eu/current/multilib/nonfree/x86_64-repodata' ...
=> gcc-multilib-4.9.2_1: removing autodeps, please wait...
=> ERROR: xbps-src: unexistent file: /void-packages/srcpkgs/glibc-32bit/template
=> ERROR: gcc-multilib-4.9.2_1: failed to transform dependency glibc-32bit
```

**Comment by beefcurtains** [comment]
@typhon Pipelight should work on amd64, we can use the cross compiler if only someone (you? me?) can get it to build.

I downloaded [binutils](https://mirrors.kernel.org/gnu/binutils/binutils-2.25.tar.bz2), mpc, mpfr and gmp from gnu ( ftp://gcc.gnu.org/pub/gcc/ ), and mingw32-w64 (for the headers) from [sourceforge](http://vorboss.dl.sourceforge.net/project/mingw-w64/mingw-w64/mingw-w64-release/mingw-w64-v4.0.2.tar.bz2).

I built mingw32-w64 with ``./configure --without-crt --prefix=/usr`` and did a ``make install DESTDIR=$PWD/build``. Then I moved the ``build/usr/include`` to ``$HOME`` as a stopgap for building gcc. So then I extracted gcc, created a build directory, created symlinks for bfs, binutils, gas, ld, opcodes (all in the binutils package), extracted the infrastructure libraries (and renamed them from $library-$version to just $library) and ran configure (in the build dir) like so:
```
CPPFLAGS=-I$HOME/include ../configure --target=x86_64-w64-mingw32 --enable-languages=c,c++
```
But when it builds ``ld``, it cannot find the required header files that should be in the mingw32-w64 package (e.g. windows.h). Using CPPFLAGS is probably the wrong way to go, but I'm not sure what would be best. Looking at the [Arch packages](https://www.archlinux.org/groups/x86_64/mingw-w64/), they seem to be buiding and packaging binutils first, as a build dependency of gcc. So that's where I'm going now.
